### PR TITLE
Unpack and rename files from PAK file to ‘Content/d2/Unpacked' directory

### LINKED ISSUE #36: Can't hear any unit-voices
i cant find the problem, they seem to have been set up properly but you cant hear any of them ingame 
they are specified in "/d2/audio/voices.yaml"
the strange thing though, is that the audiofiles for "notifications" seem to work fine

maybe @evgeniysergeev has a clue whats wrong here

### LINKED ISSUE #95: Units sounds are not working
Looking at the PAK files i have the voice definitions seems to miss a Z prefix, tho i'm not sure why it is set up like that rn, if there are different variants of the game files that has files named differently, so i decided to just open an isssue instead of PRing the changes.

### PR DESCRIPTION
**Pull request by evgeniysergeev**
Closes #95 and Closes #36 

Unpack and rename files needed, because some PAK files can contain sound files with prefix ‘Z’
This pull request will rename such files, and remove prefix ‘Z’. No other changes in yaml or engine needed for this files to play.

### CODE REVIEW

**Comment by evgeniysergeev** [comment]
List of files that should be extracted and renamed is in
```
LoadScreen:
        UnpackFiles: ZREPORT1.VOC:REPORT1.VOC,ZREPORT2.VOC:REPORT2.VOC
```

**Comment by abcdefg30** [comment]
Side note: We do this (renaming) during the content installation in the main mods. (So in theory no new code is needed.)

**Comment by evgeniysergeev** [comment]
> Side note: We do this (renaming) during the content installation in the main mods. (So in theory no new code is needed.)

I only find ability to copy some bytes from another file and save using some predefined name.
Here files can be extracted using names, not the byte offset and length.
Also, for other games it is possible to read registry to know where original data is stored, but d2 is a dos program. And we ask user to put PAK files by himself in Contents/d2 directory, there no installer at this moment for d2.
This code needed to extract data from already copied to Contents/d2 directory.

I don't see how to implement needed functionality without this pull request

File: OpenRA.Mods.D2/D2UnpackContent.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+				}
+
+				try
+				{
+					using (var stream = modData.DefaultFileSystem.Open(originalFileName))
+					{
+						string path = Platform.ResolvePath(Platform.SupportDir);
+
+						string contentPath = Path.Combine(path, "Content");
+						if (!Directory.Exists(contentPath))
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/D2UnpackContent.cs]: You could move the directory checks/creation outside of the foreach loop. Is there a need to repeat this for every file? (I think it only needs to be done once at the beginning.)

Comment (evgeniysergeev) [reply OpenRA.Mods.D2/D2UnpackContent.cs]: this is done

**Comment by evgeniysergeev** [comment]
> I see, this mod doesn't use the content installer anyway.

This will be added someday

File: OpenRA.Mods.D2/D2UnpackContent.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+							using (FileStream fs = new FileStream(newFileName, FileMode.CreateNew, FileAccess.Write))
+ 							{
+								stream.CopyTo(fs);
+								unpackedFilesCount += 1;
+								Console.WriteLine("Successfully unpacked file: {0}", fileName);
+							}
+						}
+					}
+				}
+				catch (Exception ex)
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/D2UnpackContent.cs]: This gives me `D2UnpackContent.cs(79,22): warning CS0168: The variable 'ex' is declared but never used.`.

File: OpenRA.Mods.D2/D2UnpackContent.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+				}
+				else
+				{
+					originalFileName = s.Trim();
+					fileName = originalFileName;
+				}
+
+				try
+				{
+					using (var stream = modData.DefaultFileSystem.Open(originalFileName))
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/D2UnpackContent.cs]: Is there a reason why you don't check if this file exists before trying to open it?

Comment (evgeniysergeev) [reply OpenRA.Mods.D2/D2UnpackContent.cs]: fixed

File: OpenRA.Mods.D2/D2UnpackContent.cs (hunk 2 above)
Comment (evgeniysergeev) [reply OpenRA.Mods.D2/D2UnpackContent.cs]: fixed

**Comment by evgeniysergeev** [comment]
> We usually use var instead of explicit types whenever possible, but that is not really an issue.
Looks good (and works) besides my questions/nits below.

Now vars used where possible

File: OpenRA.Mods.D2/D2UnpackContent.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The d2 mod Developers (see AUTHORS)
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/D2UnpackContent.cs]: 2019 😄

Comment (evgeniysergeev) [reply OpenRA.Mods.D2/D2UnpackContent.cs]: this will be fixed in #55

**Pull request merged**